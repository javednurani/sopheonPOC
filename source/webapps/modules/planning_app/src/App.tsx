import React from 'react';

import { AppDispatchProps, AppStateProps } from './AppContainer';
import { AppProps } from './TempSopheonShellAPI';

export type Props = AppProps<AppStateProps, AppDispatchProps>;

const App: React.FunctionComponent<Props> = () => (
  <>
    <h1>TODO</h1>
    <div>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam tellus ante, vehicula vel velit in, tincidunt venenatis augue. Nunc vulputate
      porta turpis, interdum euismod nunc. Sed interdum commodo tortor, quis consequat mi hendrerit vitae. Aliquam laoreet ac nulla ac sodales. Fusce
      id blandit turpis. Phasellus quis nisl augue. Suspendisse molestie quam sed elementum varius. Nullam augue nulla, porttitor ut nunc vel,
      hendrerit porttitor tortor. Donec elementum condimentum odio, at molestie enim iaculis vel. Praesent fringilla enim sed nisl semper finibus.
      Phasellus odio magna, interdum a accumsan tempus, vehicula vel nisl. Etiam vel vestibulum est, vitae sagittis lectus. Nulla vulputate justo ac
      diam ultricies laoreet.
    </div>
    <br />
    <div>
      Vivamus eleifend erat vitae ante aliquam lacinia. Vivamus in imperdiet ex. Donec pellentesque convallis tristique. Quisque at ligula metus.
      Proin vel posuere erat, vel eleifend diam. Nullam metus purus, egestas vel sapien eu, condimentum suscipit massa. Ut laoreet arcu neque, nec
      volutpat felis vestibulum ac. Donec ac nisl lectus. Suspendisse ultrices ligula vel erat aliquam vehicula. Class aptent taciti sociosqu ad
      litora torquent per conubia nostra, per inceptos himenaeos. Morbi sit amet massa commodo, vestibulum dolor id, vestibulum ex.
    </div>
    <br />
    <div>
      Morbi semper libero non aliquam congue. Nam sit amet lacinia augue. Quisque sollicitudin ut quam quis ultricies. Nunc ultricies nisl ac orci
      tempus feugiat. Donec eget tempor mauris. Nullam ornare tristique tincidunt. Class aptent taciti sociosqu ad litora torquent per conubia nostra,
      per inceptos himenaeos.
    </div>
    <br />
    <div>
      Sed luctus dui pellentesque mi faucibus, ac blandit ligula dapibus. Aliquam elementum accumsan nulla vitae semper. Aenean sit amet consectetur
      nulla, vitae maximus augue. Cras eu arcu odio. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Sed sed
      odio pretium, rutrum risus sit amet, elementum metus. Praesent sit amet eros et massa mattis volutpat. Ut et odio sit amet massa laoreet dapibus
      id sit amet elit.
    </div>
    <br />
    <div>
      Integer interdum sed metus at cursus. Nullam ultricies efficitur tortor ut porttitor. Donec dictum nisi non tellus tristique interdum. Sed
      consequat libero sit amet volutpat interdum. Curabitur et fringilla metus. Morbi mattis magna in quam viverra, a cursus felis luctus. Fusce at
      ultrices felis, non hendrerit justo. Cras eget urna at turpis faucibus rutrum at eget erat. Nunc aliquet rutrum molestie.
    </div>
    <br />
    <div>
      Donec tellus elit, elementum sed hendrerit eget, iaculis et sem. Cras efficitur augue ut cursus maximus. Mauris vel condimentum arcu, a
      fermentum velit. Etiam lectus velit, egestas nec interdum sed, fringilla ut urna. Ut at urna nec ex imperdiet porttitor. Lorem ipsum dolor sit
      amet, consectetur adipiscing elit. Curabitur scelerisque tellus id turpis cursus pretium. Mauris tempus, turpis id tristique lacinia, nisi neque
      tempor eros, nec viverra velit velit eget orci. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Nullam ac
      fermentum dui, vel egestas tortor. Aliquam luctus elit ut turpis suscipit, placerat blandit odio vestibulum. Fusce dapibus sodales magna,
      ultrices tristique tortor fermentum vel. Sed facilisis nisl urna, et tincidunt dolor tempor vitae. In hac habitasse platea dictumst.
    </div>
    <br />
    <div>
      Praesent sollicitudin pharetra nisi, vitae mattis ligula imperdiet condimentum. Duis sit amet nulla arcu. Morbi id eros at justo volutpat
      sodales vel non felis. Praesent varius urna nec tortor pellentesque posuere. Phasellus nisl ante, volutpat quis blandit a, tincidunt nec libero.
      Curabitur quam neque, molestie quis congue rhoncus, pulvinar ut mauris. Aliquam eget efficitur est, nec feugiat nunc. Curabitur dolor lectus,
      vehicula quis elit in, egestas consequat odio. Aliquam molestie aliquam neque, in lobortis diam efficitur eu. Nam faucibus nisi in mi venenatis,
      a scelerisque dolor convallis. Aenean ac egestas leo. Aliquam ut sapien nunc. Nam luctus nulla nec mauris pulvinar, et faucibus orci eleifend.
    </div>
    <br />
    <div>
      Ut sit amet mauris quis odio ultricies bibendum vel vitae magna. Pellentesque ante lectus, maximus vel mi in, posuere cursus urna. In posuere
      euismod diam, vitae vehicula lacus luctus at. Integer venenatis elit ipsum, sit amet consectetur nisi pretium et. Quisque id venenatis nunc, sed
      ornare dolor. Nulla euismod lacinia vehicula. Ut est lorem, bibendum et augue in, laoreet venenatis odio. Sed semper pellentesque erat, in
      tincidunt ligula pellentesque a. Integer ac massa nulla. Aliquam lacinia lacus eget dui pharetra, sed dignissim nibh vestibulum. Nam molestie
      libero at dolor sodales ultricies.
    </div>
    <br />
    <div>
      Suspendisse id mi sed sapien accumsan sollicitudin. Nam eget justo ante. Etiam purus sapien, fermentum non tellus in, aliquet venenatis sem.
      Curabitur arcu sem, interdum ac vulputate quis, cursus et risus. Donec sed erat non neque ultricies euismod et eget risus. Quisque sit amet
      consequat nulla, varius posuere dolor. Donec vitae enim sed nisi tristique mollis. Nulla nec dolor dui. Phasellus iaculis odio ac arcu iaculis
      consectetur.
    </div>
    <br />
    <div>
      Phasellus dignissim lobortis nibh, vel laoreet arcu rutrum eu. Phasellus eget eros a elit vestibulum luctus. Mauris rutrum tortor magna, eget
      imperdiet tortor volutpat varius. Etiam ac dui lacinia, commodo dolor ac, rhoncus sapien. Vivamus commodo turpis ornare efficitur scelerisque.
      Fusce ut eros et leo tincidunt aliquet. Etiam auctor condimentum gravida. Etiam nec convallis enim, eu molestie mauris. Vivamus magna mi,
      ultrices sed accumsan sed, scelerisque a lorem. Duis ut dolor imperdiet, ornare felis sed, lacinia mi. Maecenas viverra pharetra sem.
      Pellentesque consequat nisl eu vestibulum lobortis.
    </div>
    <br />
    <div>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam tellus ante, vehicula vel velit in, tincidunt venenatis augue. Nunc vulputate
      porta turpis, interdum euismod nunc. Sed interdum commodo tortor, quis consequat mi hendrerit vitae. Aliquam laoreet ac nulla ac sodales. Fusce
      id blandit turpis. Phasellus quis nisl augue. Suspendisse molestie quam sed elementum varius. Nullam augue nulla, porttitor ut nunc vel,
      hendrerit porttitor tortor. Donec elementum condimentum odio, at molestie enim iaculis vel. Praesent fringilla enim sed nisl semper finibus.
      Phasellus odio magna, interdum a accumsan tempus, vehicula vel nisl. Etiam vel vestibulum est, vitae sagittis lectus. Nulla vulputate justo ac
      diam ultricies laoreet.
    </div>
    <br />
    <div>
      Vivamus eleifend erat vitae ante aliquam lacinia. Vivamus in imperdiet ex. Donec pellentesque convallis tristique. Quisque at ligula metus.
      Proin vel posuere erat, vel eleifend diam. Nullam metus purus, egestas vel sapien eu, condimentum suscipit massa. Ut laoreet arcu neque, nec
      volutpat felis vestibulum ac. Donec ac nisl lectus. Suspendisse ultrices ligula vel erat aliquam vehicula. Class aptent taciti sociosqu ad
      litora torquent per conubia nostra, per inceptos himenaeos. Morbi sit amet massa commodo, vestibulum dolor id, vestibulum ex.
    </div>
    <br />
    <div>
      Morbi semper libero non aliquam congue. Nam sit amet lacinia augue. Quisque sollicitudin ut quam quis ultricies. Nunc ultricies nisl ac orci
      tempus feugiat. Donec eget tempor mauris. Nullam ornare tristique tincidunt. Class aptent taciti sociosqu ad litora torquent per conubia nostra,
      per inceptos himenaeos.
    </div>
    <br />
    <div>
      Sed luctus dui pellentesque mi faucibus, ac blandit ligula dapibus. Aliquam elementum accumsan nulla vitae semper. Aenean sit amet consectetur
      nulla, vitae maximus augue. Cras eu arcu odio. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Sed sed
      odio pretium, rutrum risus sit amet, elementum metus. Praesent sit amet eros et massa mattis volutpat. Ut et odio sit amet massa laoreet dapibus
      id sit amet elit.
    </div>
    <br />
    <div>
      Integer interdum sed metus at cursus. Nullam ultricies efficitur tortor ut porttitor. Donec dictum nisi non tellus tristique interdum. Sed
      consequat libero sit amet volutpat interdum. Curabitur et fringilla metus. Morbi mattis magna in quam viverra, a cursus felis luctus. Fusce at
      ultrices felis, non hendrerit justo. Cras eget urna at turpis faucibus rutrum at eget erat. Nunc aliquet rutrum molestie.
    </div>
    <br />
    <div>
      Donec tellus elit, elementum sed hendrerit eget, iaculis et sem. Cras efficitur augue ut cursus maximus. Mauris vel condimentum arcu, a
      fermentum velit. Etiam lectus velit, egestas nec interdum sed, fringilla ut urna. Ut at urna nec ex imperdiet porttitor. Lorem ipsum dolor sit
      amet, consectetur adipiscing elit. Curabitur scelerisque tellus id turpis cursus pretium. Mauris tempus, turpis id tristique lacinia, nisi neque
      tempor eros, nec viverra velit velit eget orci. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Nullam ac
      fermentum dui, vel egestas tortor. Aliquam luctus elit ut turpis suscipit, placerat blandit odio vestibulum. Fusce dapibus sodales magna,
      ultrices tristique tortor fermentum vel. Sed facilisis nisl urna, et tincidunt dolor tempor vitae. In hac habitasse platea dictumst.
    </div>
    <br />
    <div>
      Praesent sollicitudin pharetra nisi, vitae mattis ligula imperdiet condimentum. Duis sit amet nulla arcu. Morbi id eros at justo volutpat
      sodales vel non felis. Praesent varius urna nec tortor pellentesque posuere. Phasellus nisl ante, volutpat quis blandit a, tincidunt nec libero.
      Curabitur quam neque, molestie quis congue rhoncus, pulvinar ut mauris. Aliquam eget efficitur est, nec feugiat nunc. Curabitur dolor lectus,
      vehicula quis elit in, egestas consequat odio. Aliquam molestie aliquam neque, in lobortis diam efficitur eu. Nam faucibus nisi in mi venenatis,
      a scelerisque dolor convallis. Aenean ac egestas leo. Aliquam ut sapien nunc. Nam luctus nulla nec mauris pulvinar, et faucibus orci eleifend.
    </div>
    <br />
    <div>
      Ut sit amet mauris quis odio ultricies bibendum vel vitae magna. Pellentesque ante lectus, maximus vel mi in, posuere cursus urna. In posuere
      euismod diam, vitae vehicula lacus luctus at. Integer venenatis elit ipsum, sit amet consectetur nisi pretium et. Quisque id venenatis nunc, sed
      ornare dolor. Nulla euismod lacinia vehicula. Ut est lorem, bibendum et augue in, laoreet venenatis odio. Sed semper pellentesque erat, in
      tincidunt ligula pellentesque a. Integer ac massa nulla. Aliquam lacinia lacus eget dui pharetra, sed dignissim nibh vestibulum. Nam molestie
      libero at dolor sodales ultricies.
    </div>
    <br />
    <div>
      Suspendisse id mi sed sapien accumsan sollicitudin. Nam eget justo ante. Etiam purus sapien, fermentum non tellus in, aliquet venenatis sem.
      Curabitur arcu sem, interdum ac vulputate quis, cursus et risus. Donec sed erat non neque ultricies euismod et eget risus. Quisque sit amet
      consequat nulla, varius posuere dolor. Donec vitae enim sed nisi tristique mollis. Nulla nec dolor dui. Phasellus iaculis odio ac arcu iaculis
      consectetur.
    </div>
    <br />
    <div>
      Phasellus dignissim lobortis nibh, vel laoreet arcu rutrum eu. Phasellus eget eros a elit vestibulum luctus. Mauris rutrum tortor magna, eget
      imperdiet tortor volutpat varius. Etiam ac dui lacinia, commodo dolor ac, rhoncus sapien. Vivamus commodo turpis ornare efficitur scelerisque.
      Fusce ut eros et leo tincidunt aliquet. Etiam auctor condimentum gravida. Etiam nec convallis enim, eu molestie mauris. Vivamus magna mi,
      ultrices sed accumsan sed, scelerisque a lorem. Duis ut dolor imperdiet, ornare felis sed, lacinia mi. Maecenas viverra pharetra sem.
      Pellentesque consequat nisl eu vestibulum lobortis.
    </div>
  </>
);

export default App;
