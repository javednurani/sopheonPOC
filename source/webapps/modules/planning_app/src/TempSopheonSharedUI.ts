import { createTheme } from '@fluentui/react';
import { Theme } from '@fluentui/theme';
import { Reducer, Store } from 'redux';
import { Saga } from 'redux-saga';

// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
// THIS IS A TEMPORARY FILE TO SHARE FILES WITH THE TEMPLATE APP
// THESE FILES ARE NORMALLY ACCESSIBLE IN THE @sopheon/shared-ui PACKAGE
// ONCE THIS APP IS ADDED TO A SHELL WITH THAT PACKAGE
// THIS FILE SHOULD BE DELETED AND IMPORTS REPLACED WITH THE PACKAGE
// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

export interface IEnhancedStore extends Store {
  injectReducer?: (key: string, asyncReducer: Reducer<any>) => void;
  injectSaga?: (key: string, asyncSaga: Saga<any>) => void;
  asyncReducers?: { [key: string]: Reducer };
  asyncSagas?: { [key: string]: Saga };
}

const lightTheme: Theme = createTheme({
  palette: {
    themePrimary: '#72379e',
    themeLighterAlt: '#f8f5fb',
    themeLighter: '#e5d7ef',
    themeLight: '#cfb6e2',
    themeTertiary: '#a478c5',
    themeSecondary: '#8049aa',
    themeDarkAlt: '#66328e',
    themeDark: '#562a78',
    themeDarker: '#401f59',
    neutralLighterAlt: '#faf9f8',
    neutralLighter: '#f3f2f1',
    neutralLight: '#edebe9',
    neutralQuaternaryAlt: '#e1dfdd',
    neutralQuaternary: '#d0d0d0',
    neutralTertiaryAlt: '#c8c6c4',
    neutralTertiary: '#595959',
    neutralSecondary: '#373737',
    neutralPrimaryAlt: '#2f2f2f',
    neutralPrimary: '#000000',
    neutralDark: '#151515',
    black: '#0b0b0b',
    white: '#ffffff',
  },
});
lightTheme.id = 'lightTheme';

const messages: Record<string, Record<string, string>> = {
  en: {
    'app.welcome': 'welcome to app',
    'app.add5': 'Add 5',
    'app.add5_aria': 'Increment value by 5',
    'aria.increment': 'Increment',
    'aria.decrement': 'Decremement',
    'fallback.loading': 'Loading...',
  },
};

export { lightTheme, messages };
