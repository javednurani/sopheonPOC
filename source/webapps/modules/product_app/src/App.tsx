import { registerIcons, Stack } from '@fluentui/react';
import { AppProps } from '@sopheon/shell-api';
import React, { useEffect } from 'react';

import { AppDispatchProps, AppStateProps } from './AppContainer';
import { ReactComponent as AeroIndustry } from './images/industryico_Aero.svg';
import { ReactComponent as AgIndustry } from './images/industryico_Ag.svg';
import { ReactComponent as AutoIndustry } from './images/industryico_Auto.svg';
import { ReactComponent as ConstREIndustry } from './images/industryico_ConstRE.svg';
import { ReactComponent as ConsumerIndustry } from './images/industryico_Consumer.svg';
import { ReactComponent as EdIndustry } from './images/industryico_Ed.svg';
import { ReactComponent as EnergyIndustry } from './images/industryico_Energy.svg';
import { ReactComponent as FinIndustry } from './images/industryico_Fin.svg';
import { ReactComponent as GovtIndustry } from './images/industryico_Govt.svg';
import { ReactComponent as HealthIndustry } from './images/industryico_Health.svg';
import { ReactComponent as HospIndustry } from './images/industryico_Hosp.svg';
import { ReactComponent as IndusIndustry } from './images/industryico_Indus.svg';
import { ReactComponent as MediaIndustry } from './images/industryico_Media.svg';
import { ReactComponent as MemberIndustry } from './images/industryico_Member.svg';
import { ReactComponent as ServicesIndustry } from './images/industryico_Services.svg';
import { ReactComponent as TechIndustry } from './images/industryico_Tech.svg';
import { ReactComponent as TeleIndustry } from './images/industryico_Tele.svg';
import { ReactComponent as TransIndustry } from './images/industryico_Trans.svg';
import OnboardingInfo from './onboardingInfo';
import { EnvironmentScopedApiRequestDto } from './types';

export type Props = AppProps<AppStateProps, AppDispatchProps>;

const svgIconStyle: React.CSSProperties = {
  width: '20px',
  height: '20px',
  overflow: 'visible',
};
registerIcons({
  icons: {
    AgIndustryIcon: <AgIndustry style={svgIconStyle} />,
    AeroIndustryIcon: <AeroIndustry style={svgIconStyle} />,
    AutoIndustryIcon: <AutoIndustry style={svgIconStyle} />,
    ConstREIndustryIcon: <ConstREIndustry style={svgIconStyle} />,
    ConsumerIndustryIcon: <ConsumerIndustry style={svgIconStyle} />,
    EduIndustryIcon: <EdIndustry style={svgIconStyle} />,
    EnergyIndustryIcon: <EnergyIndustry style={svgIconStyle} />,
    FinIndustryIcon: <FinIndustry style={svgIconStyle} />,
    GovtIndustryIcon: <GovtIndustry style={svgIconStyle} />,
    HealthIndustryIcon: <HealthIndustry style={svgIconStyle} />,
    HospIndustryIcon: <HospIndustry style={svgIconStyle} />,
    IndusIndustryIcon: <IndusIndustry style={svgIconStyle} />,
    MediaIndustryIcon: <MediaIndustry style={svgIconStyle} />,
    MemberIndustryIcon: <MemberIndustry style={svgIconStyle} />,
    ServicesIndustryIcon: <ServicesIndustry style={svgIconStyle} />,
    TechIndustryIcon: <TechIndustry style={svgIconStyle} />,
    TeleIndustryIcon: <TeleIndustry style={svgIconStyle} />,
    TransIndustryIcon: <TransIndustry style={svgIconStyle} />,
  },
});

const App: React.FunctionComponent<Props> = ({
  currentStep,
  createProduct,
  updateProduct,
  getProducts,
  environmentKey,
  getAccessToken,
  accessToken,
}: Props) => {
  useEffect(() => {
    // getAccessToken triggers Shell action to store access token, freshly acquired from MSAL, in Redux state
    // after getAccessToken is called (here, on ProductApp render), shellApi::accessToken should be up-to-date access token from MSAL.acquireTokenSilent()
    getAccessToken();

    // get any Products for User on initial load of Product App (will determine Onboarding State, ____, etc)
    const requestDto: EnvironmentScopedApiRequestDto = {
      EnvironmentKey: environmentKey,
      AccessToken: accessToken,
    };

    getProducts(requestDto);
  }, []);

  return (
    <div>
      <Stack>
        <OnboardingInfo
          currentStep={currentStep}
          createProduct={createProduct}
          updateProduct={updateProduct}
          environmentKey={environmentKey}
          accessToken={accessToken}
        />
      </Stack>
    </div>
  );
};
export default App;
