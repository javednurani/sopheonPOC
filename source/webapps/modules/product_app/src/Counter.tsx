import { IconButton, IIconProps, PrimaryButton, Stack } from '@fluentui/react';
import React, { CSSProperties } from 'react';
import { useIntl } from 'react-intl';

import { AppDispatchProps, AppStateProps } from './AppContainer';

export type CounterProps = AppStateProps & AppDispatchProps;

const Counter: React.FunctionComponent<CounterProps> = ({
  counterValue,
  decrementCounter,
  incrementCounter,
  incrementCounterAsync,
}: CounterProps) => {
  const { formatMessage } = useIntl();

  const addIcon: IIconProps = { iconName: 'Add' };
  const subtractIcon: IIconProps = { iconName: 'Remove' };

  const countStyles: CSSProperties = {
    fontSize: '48px',
  };

  return (
    <Stack className="counterContainer" horizontalAlign="center">
      <span id="counterValue" style={countStyles}>
        {counterValue}
      </span>
      <Stack horizontal>
        <IconButton iconProps={addIcon} aria-label={formatMessage({ id: 'aria.increment' })} onClick={() => incrementCounter()} />
        <IconButton iconProps={subtractIcon} aria-label={formatMessage({ id: 'aria.decrement' })} onClick={() => decrementCounter()} />
        <PrimaryButton
          text={formatMessage({ id: 'app.add5' })}
          aria-label={formatMessage({ id: 'app.add5_aria' })}
          onClick={() => incrementCounterAsync()}
        />
      </Stack>
    </Stack>
  );
};

export default Counter;
