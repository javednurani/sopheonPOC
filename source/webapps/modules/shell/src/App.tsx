import { initializeIcons, registerIcons, ScrollablePane, ScrollbarVisibility, Stack } from '@fluentui/react';
import { useTheme } from '@fluentui/react-theme-provider';
import React, { CSSProperties, FunctionComponent } from 'react';
import { useIntl } from 'react-intl';
import { BrowserRouter, Route, Switch } from 'react-router-dom';

import IdleMonitor from './authentication/IdleMonitor';
import Login from './authentication/Login';
import Signup from './authentication/Signup';
import { DynamicModule } from './DynamicModule';
import Footer from './footer/Footer';
import Header from './header/Header';
import { ReactComponent as LLogo } from './images/Lucy24_logo.svg';
import { appModules } from './settings/appModuleSettings';
import { shellApi } from './ShellApi';
import { ChangeThemeAction } from './themes/themeReducer/themeReducer';

export interface AppProps {
  changeTheme: (useDarkTheme: boolean) => ChangeThemeAction;
}

const App: FunctionComponent<AppProps> = ({ changeTheme }: AppProps) => {
  const { formatMessage } = useIntl();
  const loadingMessage: string = formatMessage({ id: 'fallback.loading' });
  useTheme();

  initializeIcons();

  const lucyIconStyle: CSSProperties = {
    width: '20px',
    height: '20px',
    overflow: 'visible',
  };

  registerIcons({
    icons: {
      Lucy: <LLogo style={lucyIconStyle} />,
    },
  });

  const appStyle: CSSProperties = {
    textAlign: 'center',
    height: '100vh',
  };

  const pageContainerStyle: CSSProperties = {
    position: 'relative',
    height: '100%',
  };

  // TODO: check for user product existance CLOUD-2148
  const userHasProduct = false;

  return (
    <div className="App" style={appStyle}>
      <BrowserRouter>
        <Switch>
          <Route exact path="/login">
            <Login />
          </Route>
          <Route exact path="/signup">
            <Signup />
          </Route>
          <Route path="/">
            <Stack
              grow
              styles={{
                root: {
                  height: '100%',
                  width: '100%',
                },
              }}
            >
              {(!location.pathname.includes('product') || userHasProduct) &&
                <Stack.Item>
                  <Header changeTheme={changeTheme} />
                </Stack.Item>
              }
              <Stack.Item shrink>
                <IdleMonitor />
              </Stack.Item>
              <Stack.Item
                verticalFill
                styles={{
                  root: {
                    height: '100%',
                    overflow: 'auto',
                  },
                }}
              >
                <main style={pageContainerStyle} role="main">
                  <ScrollablePane scrollbarVisibility={ScrollbarVisibility.auto}>
                    <Switch>
                      {appModules.map(appModule => (
                        <Route exact key={appModule.scope} path={appModule.routeName}>
                          <DynamicModule module={appModule} loadingMessage={loadingMessage} shellApi={shellApi} />
                        </Route>
                      ))}
                    </Switch>
                  </ScrollablePane>
                </main>
              </Stack.Item>
              {(!location.pathname.includes('product') || userHasProduct) &&
                <Stack.Item>
                  <Footer />
                </Stack.Item>
              }
            </Stack>
          </Route>
        </Switch>
      </BrowserRouter>
    </div>
  );
};

export default App;
