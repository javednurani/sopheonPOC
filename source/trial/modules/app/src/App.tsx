import { Label } from '@fluentui/react';
import { AppProps } from '@sopheon/shell-api';
import React from 'react';
import { FormattedMessage } from 'react-intl';

import { AppDispatchProps, AppStateProps } from './AppContainer';
import Counter from './Counter';

export type Props = AppProps<AppStateProps, AppDispatchProps>;

const App: React.FunctionComponent<Props> = ({ counterValue, decrementCounter, incrementCounter, incrementCounterAsync }: Props) => (
  <div>
    <Label>
      <FormattedMessage id="app.welcome" />
    </Label>
    <Counter
      counterValue={counterValue}
      decrementCounter={decrementCounter}
      incrementCounter={incrementCounter}
      incrementCounterAsync={incrementCounterAsync}
    />
  </div>
);

export default App;
