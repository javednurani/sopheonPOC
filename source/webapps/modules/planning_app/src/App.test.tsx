import { Label } from '@fluentui/react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { axe, toHaveNoViolations } from 'jest-axe';
import React from 'react';
import { IntlProvider } from 'react-intl';

import App, { Props } from './App';
import Counter from './Counter';
import { messages } from './TempSopheonSharedUI';

expect.extend(toHaveNoViolations);

describe('Testing the App component', () => {
  it('Render test for the App component', () => {
    const appProps: Props = {
      counterValue: 1,
      decrementCounter: jest.fn(),
      incrementCounter: jest.fn(),
      incrementCounterAsync: jest.fn(),
    };

    const wrapper = mount(
      <IntlProvider locale="en" messages={messages.en}>
        <App
          counterValue={appProps.counterValue}
          decrementCounter={appProps.decrementCounter}
          incrementCounter={appProps.incrementCounter}
          incrementCounterAsync={appProps.incrementCounterAsync}
        />
      </IntlProvider>
    );
    expect(wrapper.find(Label)).toHaveLength(1);
    expect(wrapper.find(Label).text()).toBe(messages.en['app.welcome']);
    expect(wrapper.find(Counter)).toHaveLength(1);
    expect(wrapper.find(Counter).props()).toStrictEqual(appProps);
  });
  it('Accessibility test for the App component', async () => {
    const appProps: Props = {
      counterValue: 1,
      decrementCounter: jest.fn(),
      incrementCounter: jest.fn(),
      incrementCounterAsync: jest.fn(),
    };

    const wrapper = mount(
      <IntlProvider locale="en" messages={messages.en}>
        <App
          counterValue={appProps.counterValue}
          decrementCounter={appProps.decrementCounter}
          incrementCounter={appProps.incrementCounter}
          incrementCounterAsync={appProps.incrementCounterAsync}
        />
      </IntlProvider>
    );
    const results = await axe(wrapper.getDOMNode(), {
      rules: {
        // These axe rules are fixed when app is run in the shell
        region: { enabled: false },
      },
    });
    expect(results).toHaveNoViolations();
  });
  it('App snapshot render test', () => {
    const appProps: Props = {
      counterValue: 1,
      decrementCounter: jest.fn(),
      incrementCounter: jest.fn(),
      incrementCounterAsync: jest.fn(),
    };

    const tree = mount(
      <IntlProvider locale="en" messages={messages.en}>
        <App {...appProps} />
      </IntlProvider>
    );

    expect(
      toJson(tree, {
        noKey: false,
        mode: 'deep',
      })
    ).toMatchSnapshot();
  });
});
