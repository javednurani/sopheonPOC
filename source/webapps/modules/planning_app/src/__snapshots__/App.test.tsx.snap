// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the App component App snapshot render test 1`] = `
<div>
  <label
    className="ms-Label root-40"
  >
    welcome to app
  </label>
  <div
    className="ms-Stack counterContainer css-41"
  >
    <span
      id="counterValue"
      key=".0:$.0"
      style={
        Object {
          "fontSize": "48px",
        }
      }
    >
      1
    </span>
    <div
      className="ms-Stack css-42"
    >
      Array [
        <button
          aria-label="Increment"
          className="ms-Button ms-Button--icon root-43"
          data-is-focusable={true}
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          type="button"
        >
          <span
            className="ms-Button-flexContainer flexContainer-44"
            data-automationid="splitbuttonprimary"
          >
            <i
              aria-hidden={true}
              className="ms-Icon root-37 ms-Button-icon icon-46"
              data-icon-name="Add"
              role="presentation"
              style={
                Object {
                  "fontFamily": undefined,
                }
              }
            />
          </span>
        </button>,
        "",
      ]
      Array [
        <button
          aria-label="Decremement"
          className="ms-Button ms-Button--icon root-43"
          data-is-focusable={true}
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          type="button"
        >
          <span
            className="ms-Button-flexContainer flexContainer-44"
            data-automationid="splitbuttonprimary"
          >
            <i
              aria-hidden={true}
              className="ms-Icon root-37 ms-Button-icon icon-46"
              data-icon-name="Remove"
              role="presentation"
              style={
                Object {
                  "fontFamily": undefined,
                }
              }
            />
          </span>
        </button>,
        "",
      ]
      Array [
        <button
          aria-label="Increment value by 5"
          className="ms-Button ms-Button--primary root-51"
          data-is-focusable={true}
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          type="button"
        >
          <span
            className="ms-Button-flexContainer flexContainer-44"
            data-automationid="splitbuttonprimary"
          >
            <span
              className="ms-Button-textContainer textContainer-45"
            >
              <span
                className="ms-Button-label label-52"
                id="id__24"
                key="id__24"
              >
                Add 5
              </span>
            </span>
          </span>
        </button>,
        "",
      ]
    </div>
  </div>
</div>
`;
